public/js/loginValidation.js

```diff
@@ -0,0 +1,62 @@
+
+function isValidEmail(elementID) {
+    let inputVal = document.getElementById(elementID);
+    let email = inputVal.value.trim();
+    let re = new RegExp("^\\w+([\\.-]?\\w+)*@\\w+([\\.-]?\\w+)*(\\.\\w{2,3})+$");
+
+    if (re.test(email)) {
+        inputVal.style.backgroundColor = "";
+        return true
+    } else {
+        inputVal.style.backgroundColor = "pink";
+        return false
+    }
+}
+
+function checkRequiredField(elementID) {
+    let inputVal = document.getElementById(elementID);
+
+    if (inputVal.value.trim() == "") {
+        inputVal.style.backgroundColor = "pink";
+        return false;
+    }
+    else{
+        inputVal.style.backgroundColor = "";
+        return true;
+    }
+}
+
+
+
+function validateForm() {
+    let cleanedEmailInput = encodeURI(document.getElementById('userEmail').value);
+    let cleanedPasswordInput = encodeURI(document.getElementById('password').value);
+
+    if (checkRequiredField('userEmail') && isValidEmail('userEmail') && checkRequiredField('password')) {
+        document.getElementById('submitWarning').textContent = 'Form is Goooood!'
+
+        fetch("/api/login",
```

Comment:
function is called 'validate form' but this is also sending AJAX

Reply:
change to validateToPostInputs

Reply:
Function is currently doing 2 things but maybe should be doing one? One to cleanse and one to send data that gets called inside the cleanse. Nick has an idea for it - see nick